packages/remote-feature-flag-controller/src/remote-feature-flag-controller.ts

```diff
@@ -291,14 +332,43 @@ export class RemoteFeatureFlagController extends BaseController<
         continue;
       }
 
-      if (Array.isArray(processedValue) && thresholdValue) {
+      if (Array.isArray(processedValue)) {
+        // Validate array has valid threshold items before doing expensive crypto operation
+        const hasValidThresholds = processedValue.some(
+          isFeatureFlagWithScopeValue,
+        );
+
+        if (!hasValidThresholds) {
+          // Not a threshold array - preserve as-is
+          processedRemoteFeatureFlags[remoteFeatureFlagName] = processedValue;
+          continue;
+        }
+
+        // Check cache first, calculate only if needed
+        const cacheKey: `${string}:${string}` = `${metaMetricsId}:${remoteFeatureFlagName}`;
```

Comment:
Nit: This should also work:
```suggestion
        const cacheKey = `${metaMetricsId}:${remoteFeatureFlagName}` as const;
```